model-optimizer/extensions/middle/L2NormToNorm.py

```diff
@@ -82,6 +82,27 @@ def replace_pattern(self, graph: Graph, match: dict):
             log.debug('The value of the "maximum_y_data" is not defined or is not constant')
             return
 
+        # We need to check axes which performed reduction because IE supports only 2D, 3D, 4D inputs and
+        # reduction only along spatial dimensions.
+        input_shape = match['sum'].in_port(0).data.get_shape().size
```

Comment:
Maybe `input_rank`, not `input_shape`?

Reply:
Yes, it is better name. I will change this name if there will be another change requests in this PR.

Feel free to reopen this discussion if you think what this change is necessary.